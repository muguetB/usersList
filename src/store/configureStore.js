// @flow

import createSagaMiddleware from 'redux-saga'
import { applyMiddleware, combineReducers, createStore, compose } from 'redux'
import createReducers from '../redux'
import sagas from '../sagas'

export default function configureStore() {
  const sagaMiddleware = createSagaMiddleware()

  const initialState =
    window &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()

  const middleware = applyMiddleware(sagaMiddleware)

  const appReducer = combineReducers(createReducers())

  const rootReducer = (state, action) => appReducer(state, action)

  const store = createStore(rootReducer, initialState, compose(middleware))

  sagas.forEach(saga => saga.map(sagaMiddleware.run))

  return store
}
