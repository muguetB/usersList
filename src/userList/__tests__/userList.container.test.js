import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import UserListContainer from '../userList.container.js'
import React from 'react'

describe('UserList container', () => {
  it('should render correctly', () => {
    // given
    const props = {
      nextPageExist: false,
    }

    const mockStore = configureStore()({
      usersList: {
        users: [],
        loading: false,
        error: false,
        emptyUsersList: false,
      },
    })

    // when
    const wrapper = renderer.create(
      <Provider store={mockStore}>
        <UserListContainer {...props} />
      </Provider>,
    )

    // then
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
