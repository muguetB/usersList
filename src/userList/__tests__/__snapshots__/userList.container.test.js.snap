// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList container should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingTop": 45,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Jeste≈õ na stronie 
      1
    </Text>
  </View>
  <View>
    <Text>
      Waiting for data
    </Text>
  </View>
</View>
`;
